app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -37,20 +37,10 @@ def index_for_group
     render json: roles
   end
 
-  private def roles_of_group_type(group_type)
-    group_ids = UserGroup.where(group_type: group_type).pluck(:id)
-    UserRole.where(group_id: group_ids)
-  end
-
   # Returns a list of roles primarily based on groupType.
   def index_for_group_type
     group_type = params.require(:group_type)
-    roles = roles_of_group_type(group_type)
-
-    # Filter & Sort roles
-    roles = UserRole.filter_roles(roles, current_user, params)
-    roles = UserRole.sort_roles(roles, params[:sort])
-
+    roles = UserGroup.roles_of_group_type(group_type, current_user, params)
```

Comment:
See my remark about using `roles_of_group_type` as a "god method" above.
You are of course free to _also_ provide an _additional_ "does it all" method as a convenience, but the basic `roles_of_group_type` should not do filtering and sorting.

Reply:
Okay done.

Reply:
Okay done